proposals/NNNN-migration-tooling-for-upcoming-features.md

```diff
@@ -0,0 +1,124 @@
+# Migration tooling for Swift upcoming features
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: TBD
+* Review: TBD
+
+## Introduction
+
+As Swift evolves, source-breaking changes to the language are staged behind new
+language modes and, more recently, [upcoming features][SE-0362] that may be
+enabled piecemeal without adopting an entire new language mode. 
+
+In this proposal, we zoom in on the experience of adopting *individual* 
+source-breaking upcoming features. The premise is that, in principle, migration 
+is generally separable from adoption (?), and that decoupling migration will 
+empower compiler engineers to offer significantly richer diagnostics and 
+guidance that would not be appropriate in regular builds.
+
+[SE-0362]: https://github.com/swiftlang/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md
+
+## Motivation
```

Comment:
I think the primary motivation that's missing (at least from my read) is that for many upcoming features, it is possible to automatically offer source changes to make the code work when the upcoming feature is enabled.